src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -95,18 +103,44 @@ public function getItem(string $resourceClass, $id, string $operationName = null
         $repository = $manager->getRepository($resourceClass);
         $queryBuilder = $repository->createQueryBuilder('o');
 
-        foreach ($identifiers as $propertyName => $value) {
-            $placeholder = 'id_'.$propertyName;
-
-            $queryBuilder
-                ->where($queryBuilder->expr()->eq('o.'.$propertyName, ':'.$placeholder))
-                ->setParameter($placeholder, $value);
-        }
+        $this->processIdentifiers($identifiers, $queryBuilder);
 
         foreach ($this->itemExtensions as $extension) {
             $extension->applyToItem($queryBuilder, $resourceClass, $identifiers, $operationName);
         }
 
         return $queryBuilder->getQuery()->getOneOrNullResult();
     }
+
+    /**
+     * Add where into the query when multiple identifiers are passed (composite).
+     *
+     * @param array        $identifiers
+     * @param QueryBuilder $queryBuilder
+     *
+     * Populate the query with where when needed.
+     */
+    private function processIdentifiers(array $identifiers, QueryBuilder $queryBuilder)
+    {
+        if (0 === count($identifiers)) {
+            return;
+        }
+
+        $firstIdentifier = reset($identifiers);
+        foreach ($identifiers as $identifier => $value) {
+            $placeholder = ':id_'.$identifier;
+            $expression = $queryBuilder->expr()->eq(
+                'o.'.$identifier,
+                $placeholder
+            );
+
+            if ($firstIdentifier === $identifier) {
+                $queryBuilder->where($expression);
+            } else {
+                $queryBuilder->andWhere($expression);
```

Comment:
IIRC, you can use `andWhere()` directly without testing if it's the first call or not.